src/lib/createEventClass.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @module seng-event
+ */
+import BaseEvent from './BaseEvent';
+
+/**
+ * @ignore
+ */
+type TypeMap<TType extends string> = { [P in TType]: P };
+
+/**
+ * @ignore
+ */
+interface EventTypeClass<TData, TType extends string> {
+  types: TypeMap<TType>;
+  new (type: TType, data: TData): BaseEvent<TData, TType>;
+}
+
+/**
+ * Utility function to generate a class that extends [[AbstractEvent]] and optionally has
```

Comment:
`AbstractEvent ` should be `BaseEvent` (looking at the code)

Reply:
`BaseEvent` extends `AbstractEvent` and I thought `AbstractEvent` might make more sense in the documentation, because `BaseEvent` isn't meant to be used directly and not exported from the main module (only available under `seng-event/lib/BaseEvent`). There's not really a good reason to 'hide' `BaseEvent` like this, other than that I thought having too many similar base classes may be a bit confusing.